src/FileFormats.PE/PEFile.cs

```diff
@@ -28,13 +29,14 @@ public class PEFile
         private const int PESignatureOffsetLocation = 0x3C;
         private const uint ExpectedPESignature = 0x00004550;    // PE00
 
-        public PEFile(IAddressSpace fileAddressSpace)
+        public PEFile(IAddressSpace fileAddressSpace, ulong position = 0)
         {
+            _position = position;
             _fileAddressSpace = fileAddressSpace;
```

Comment:
Rather than convert all the addresses to '_position + X' you can define the _fileAddressSpace = new RelativeAddressSpace(fileAddressSpace, position)